### PR TITLE
Fixes for #429 & #430

### LINKED ISSUE #429: ludown: -v param conflict
**Describe the bug**
`ludown -v` naturally returns the version. However, `ludown parse ToLuis` documentation accepts a parameter `-v` or `--luis_versionId` that would set the luis app version in the generated JSON as per doc below. The -v parameter as described conflicts with the -v parameter for version. The obvious workaround is to use the --luis_versionId and it works fine.

```
  Usage: ludown parse ToLuis --in <luFile> | --lu_folder <inputFolder> [-s]

  Convert .lu file(s) into LUIS JSON file. You can optionally also request LUIS batch test input file

  Options:

    --in <luFile>                          .lu file to parse
    -l, --lu_folder <inputFolder>          [Optional] Folder that has the .lu file. By default ludown will only look at the current folder. To look at all subfolders, include -s
    -o, --out_folder <outputFolder>        [Optional] Output folder for all files the tool will generate
    -s, --subfolder                        [Optional] Include sub-folders as well when looking for .lu files
    -n, --luis_name <luis_appName>         [Optional] LUIS app name
    -d, --luis_desc <luis_appDesc>         [Optional] LUIS app description
    -v, --luis_versionId <luis_versionId>  [Optional] LUIS app version (default: 0.1)
    -c, --luis_culture <luis_appCulture>   [Optional] LUIS app culture (default: en-us)
    -t, --write_luis_batch_tests           [Optional] Write out LUIS batch test json file
    --verbose                              [Optional] Get verbose messages from parser
    -h, --help                             output usage information

```

**To Reproduce**
1. Create a basic lu file. I'm using the one below.
2. Run: `ludown parse ToLuis --in model.lu -n Name -d Description -v V1 -c en-US`

test lu file:
```
> # Intent definitions

## A
- hello
- hi
- hiya

## B
- this is not good
- this is fake
- i don't like this

```
**Expected behavior**
ludown should work and generate the required json file. However, the output is:

```
PS C:\dev\luis-versioning> ludown parse ToLuis --in model.lu -n Name -d Description -v V1 -c en-US
1.0.35
```

### PR DESCRIPTION
**Pull request by vishwacsena**
Fixes 
#429 
#430

### CODE REVIEW

File: packages/Ludown/lib/parser.js [hunk 1]
```diff
@@ -88,8 +88,9 @@ const writeOutFiles = function(program,finalLUISJSON,finalQnAJSON, finalQnAAlter
         finalLUISJSON.name = program.luis_name,
         finalLUISJSON.desc = program.luis_desc;
         finalLUISJSON.culture = program.luis_culture;
-        finalQnAJSON.name = program.qna_name;
     }
+
+    if(finalQnAJSON) finalQnAJSON.name = program.qna_name;
```
Comment (cleemullins) [inline packages/Ludown/lib/parser.js]: whitespace
if (finalQnAJson)

File: packages/Ludown/lib/parser.js [hunk 2]
```diff
@@ -115,17 +116,25 @@ const writeOutFiles = function(program,finalLUISJSON,finalQnAJSON, finalQnAAlter
     }
     
     if(!program.lOutFile) {
-        if(!program.luis_name) {
-            program.lOutFile = path.basename(rootFile, path.extname(rootFile)) + "_LUISApp.json";  
+        if(program.out) {
+            program.lOutFile = program.out.includes('.')?program.out:program.out + ".json"
```
Comment (cleemullins) [inline packages/Ludown/lib/parser.js]: whitespace around ? and :

File: packages/Ludown/lib/parser.js [hunk 3]
```diff
@@ -115,17 +116,25 @@ const writeOutFiles = function(program,finalLUISJSON,finalQnAJSON, finalQnAAlter
...
+            } else {
+                program.lOutFile = program.luis_name.includes('.')?program.luis_name:program.luis_name + ".json";
+            }
         }
     }
     if(!program.qOutFile) {
-        if(!program.qna_name) {
-            program.qOutFile = path.basename(rootFile, path.extname(rootFile)) + "_qnaKB.json";
+        if(program.out) {
+            program.qOutFile = program.out.includes('.')?program.out:program.out + ".json"
```
Comment (cleemullins) [inline packages/Ludown/lib/parser.js]: whitespace

File: packages/Ludown/lib/parser.js [hunk 4]
```diff
@@ -115,17 +116,25 @@ const writeOutFiles = function(program,finalLUISJSON,finalQnAJSON, finalQnAAlter
...
-        if(!program.luis_name) {
-            program.lOutFile = path.basename(rootFile, path.extname(rootFile)) + "_LUISApp.json";  
+        if(program.out) {
+            program.lOutFile = program.out.includes('.')?program.out:program.out + ".json"
         } else {
-            program.lOutFile = program.luis_name.includes('.')?program.luis_name:program.luis_name + ".json";
+            if(!program.luis_name) {
+                program.lOutFile = path.basename(rootFile, path.extname(rootFile)) + "_LUISApp.json";  
+            } else {
+                program.lOutFile = program.luis_name.includes('.')?program.luis_name:program.luis_name + ".json";
```
Comment (cleemullins) [inline packages/Ludown/lib/parser.js]: whitespace

**Comment by vishwacsena** [comment]
@cleemullins - pushed up fixes for PR comments. Existing tests are passing. 

image

**Pull request merged**